azafea/event_processors/endless/activation/v1/migrations/81cf792f49b2_remove_region_city.py

```diff
@@ -0,0 +1,48 @@
+# type: ignore
+
+"""Remove region and city and add constraints on latitude and longitude
+
+Revision ID: 81cf792f49b2
+Revises: 3ce72dccdef8
+Create Date: 2021-06-17 12:49:31.415945
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '81cf792f49b2'
+down_revision = '3ce72dccdef8'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.drop_column('activation_v1', 'city')
+    op.drop_column('activation_v1', 'region')
+
+    op.execute(
+        'UPDATE activation_v1 \n'
+        'SET latitude = CASE WHEN latitude=90 THEN 89.5 ELSE floor(latitude)+0.5 END;'
+    )
```

Comment:
In the past when rewriting every row of a table in this way we have sometimes done this as a separate command which processes the table in batches. For example, https://github.com/endlessm/azafea/blob/c146d9b97ef8ead11ff6ae083f9b50de97c6ce3f/azafea/event_processors/endless/ping/v1/cli.py#L97-L124 fills in some generated data in a bunch of columns, and does so in batches of 5000 rows, presumably to avoid holding a lock on the whole table for a long time. There are 1 650 991 rows in this table in the production database. You could use the writable test database to see how long it would take to execute this migration, keeping the table locked. Maybe it is fast enough that we can just do it this simple way.

Reply:
I tested, if you put both statements in one, it took less than a minute

Reply:
Sounds good.